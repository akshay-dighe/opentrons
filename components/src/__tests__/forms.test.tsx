// form component tests
import React from 'react'
import Renderer from 'react-test-renderer'

import {
  DeprecatedCheckboxField,
  DropdownField,
  FormGroup,
  InputField,
  RadioGroup,
} from '..'

describe('DeprecatedCheckboxField', () => {
  it('renders correctly when unchecked', () => {
    const tree = Renderer.create(
      <DeprecatedCheckboxField
        onChange={jest.fn()}
        label="Check Box 1"
        className="foo"
      />
    ).toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('renders correctly when checked', () => {
    const tree = Renderer.create(
      <DeprecatedCheckboxField
        onChange={jest.fn()}
        label="Check Box 1"
        className="foo"
        value
      />
    ).toJSON()

    expect(tree).toMatchSnapshot()
  })
})

describe('DropdownField', () => {
  it('renders correctly with a value', () => {
    const tree = Renderer.create(
      <DropdownField
        onChange={jest.fn()}
        className="foo"
        value={'rna'}
        options={[
          { name: 'DNA', value: 'dna' },
          { name: 'RNA', value: 'rna' },
          { name: 'Protein', value: 'protein' },
        ]}
      />
    ).toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('renders correctly with a falsey value', () => {
    const tree = Renderer.create(
      <DropdownField
        onChange={jest.fn()}
        className="foo"
        value={null}
        options={[
          { name: 'DNA', value: 'dna' },
          { name: 'RNA', value: 'rna' },
          { name: 'Protein', value: 'protein' },
        ]}
      />
    ).toJSON()

    expect(tree).toMatchSnapshot()
  })
})

describe('FormGroup', () => {
  it('renders correctly', () => {
    const tree = Renderer.create(
      <FormGroup className="foo" label="This is the label">
        <div>Hey test here</div>
        <div>More test here</div>
      </FormGroup>
    ).toJSON()

    expect(tree).toMatchSnapshot()
  })
})

describe('InputField', () => {
  it('renders correctly', () => {
    const tree = Renderer.create(
      <InputField
        label="Input field"
        placeholder="Placeholder Text"
        value={null}
        units="Î¼L"
      />
    ).toJSON()

    expect(tree).toMatchSnapshot()
  })
})

describe('RadioGroup', () => {
  it('renders correctly with no checked value', () => {
    const tree = Renderer.create(
      <RadioGroup
        onChange={jest.fn()}
        options={[
          { name: 'Hazelnut', value: 'hazelnut' },
          { name: 'Chocolate', value: 'chocolate' },
          { name: 'Ginger', value: 'ginger' },
        ]}
      />
    ).toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('renders correctly with checked value', () => {
    const tree = Renderer.create(
      <RadioGroup
        onChange={jest.fn()}
        value="chocolate"
        options={[
          { name: 'Hazelnut', value: 'hazelnut' },
          { name: 'Chocolate', value: 'chocolate' },
          { name: 'Ginger', value: 'ginger' },
        ]}
      />
    ).toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('renders correctly inline', () => {
    const tree = Renderer.create(
      <RadioGroup
        onChange={jest.fn()}
        value={'chocolate'}
        options={[
          { name: 'Hazelnut', value: 'hazelnut' },
          { name: 'Chocolate', value: 'chocolate' },
          { name: 'Ginger', value: 'ginger' },
        ]}
      />
    ).toJSON()

    expect(tree).toMatchSnapshot()
  })
})
