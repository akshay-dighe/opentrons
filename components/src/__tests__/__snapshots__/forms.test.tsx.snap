// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeprecatedCheckboxField renders correctly when checked 1`] = `
.c0 {
  width: 100%;
}

.c1.spin {
  -webkit-animation: GLFYz 0.8s steps(8) infinite;
  animation: GLFYz 0.8s steps(8) infinite;
  -webkit-transform-origin: center;
  -ms-transform-origin: center;
  transform-origin: center;
}

<label
  className="form_field foo"
>
  <div
    className="checkbox_icon"
  >
    <svg
      aria-hidden="true"
      className="c0 c1"
      fill="currentColor"
      version="1.1"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M10 17l-5-5 1.4-1.4 3.6 3.6 7.6-7.6L19 8m0-5H5a2 2 0 0 0-2 2v14a2 2 0 0 0 2 2h14a2 2 0 0 0 2-2V5a2 2 0 0 0-2-2z"
        fillRule="evenodd"
      />
    </svg>
  </div>
  <input
    checked={true}
    className="input_field accessibly_hidden"
    onChange={[MockFunction]}
    type="checkbox"
  />
  <div
    className="label_text"
  >
    Check Box 1
  </div>
</label>
`;

exports[`DeprecatedCheckboxField renders correctly when unchecked 1`] = `
.c0 {
  width: 100%;
}

.c1.spin {
  -webkit-animation: GLFYz 0.8s steps(8) infinite;
  animation: GLFYz 0.8s steps(8) infinite;
  -webkit-transform-origin: center;
  -ms-transform-origin: center;
  transform-origin: center;
}

<label
  className="form_field foo"
>
  <div
    className="checkbox_icon"
  >
    <svg
      aria-hidden="true"
      className="c0 c1"
      fill="currentColor"
      version="1.1"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M19 3H5a2 2 0 0 0-2 2v14a2 2 0 0 0 2 2h14a2 2 0 0 0 2-2V5a2 2 0 0 0-2-2m0 2v14H5V5h14z"
        fillRule="evenodd"
      />
    </svg>
  </div>
  <input
    checked={false}
    className="input_field accessibly_hidden"
    onChange={[MockFunction]}
    type="checkbox"
  />
  <div
    className="label_text"
  >
    Check Box 1
  </div>
</label>
`;

exports[`DropdownField renders correctly with a falsey value 1`] = `
.c0 {
  width: 100%;
}

.c1.spin {
  -webkit-animation: GLFYz 0.8s steps(8) infinite;
  animation: GLFYz 0.8s steps(8) infinite;
  -webkit-transform-origin: center;
  -ms-transform-origin: center;
  transform-origin: center;
}

<div
  className="foo"
>
  <div
    className="dropdown_field"
  >
    <select
      className="dropdown"
      onChange={[MockFunction]}
      value=""
    >
      <option
        disabled={false}
        value=""
      >
        
      </option>
      <option
        disabled={false}
        value="dna"
      >
        DNA
      </option>
      <option
        disabled={false}
        value="rna"
      >
        RNA
      </option>
      <option
        disabled={false}
        value="protein"
      >
        Protein
      </option>
    </select>
    <div
      className="dropdown_icon"
    >
      <svg
        aria-hidden="true"
        className="c0 c1"
        fill="currentColor"
        version="1.1"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M7,10L12,15L17,10H7Z"
          fillRule="evenodd"
        />
      </svg>
    </div>
  </div>
  <div
    className="input_caption"
  >
    <span />
  </div>
</div>
`;

exports[`DropdownField renders correctly with a value 1`] = `
.c0 {
  width: 100%;
}

.c1.spin {
  -webkit-animation: GLFYz 0.8s steps(8) infinite;
  animation: GLFYz 0.8s steps(8) infinite;
  -webkit-transform-origin: center;
  -ms-transform-origin: center;
  transform-origin: center;
}

<div
  className="foo"
>
  <div
    className="dropdown_field"
  >
    <select
      className="dropdown"
      onChange={[MockFunction]}
      value="rna"
    >
      <option
        disabled={false}
        value="dna"
      >
        DNA
      </option>
      <option
        disabled={false}
        value="rna"
      >
        RNA
      </option>
      <option
        disabled={false}
        value="protein"
      >
        Protein
      </option>
    </select>
    <div
      className="dropdown_icon"
    >
      <svg
        aria-hidden="true"
        className="c0 c1"
        fill="currentColor"
        version="1.1"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M7,10L12,15L17,10H7Z"
          fillRule="evenodd"
        />
      </svg>
    </div>
  </div>
  <div
    className="input_caption"
  >
    <span />
  </div>
</div>
`;

exports[`FormGroup renders correctly 1`] = `
<div
  className="foo"
>
  <div
    className="form_group_label"
  >
    This is the label
  </div>
  <div>
    Hey test here
  </div>
  <div>
    More test here
  </div>
</div>
`;

exports[`InputField renders correctly 1`] = `
<label
  className="form_field"
>
  <div
    className="label_text"
  >
    Input field
  </div>
  <div
    className="input_field_container"
  >
    <div
      className="input_field"
    >
      <input
        placeholder="Placeholder Text"
        type="text"
        value=""
      />
      <div
        className="suffix"
      >
        Î¼L
      </div>
    </div>
    <div
      className="input_caption"
    >
      <span />
      <span
        className="right"
      />
    </div>
  </div>
</label>
`;

exports[`RadioGroup renders correctly inline 1`] = `
.c0 {
  width: 100%;
}

.c1.spin {
  -webkit-animation: GLFYz 0.8s steps(8) infinite;
  animation: GLFYz 0.8s steps(8) infinite;
  -webkit-transform-origin: center;
  -ms-transform-origin: center;
  transform-origin: center;
}

<div
  className=""
>
  <label
    className="form_field"
  >
    <div
      className="checkbox_icon"
    >
      <svg
        aria-hidden="true"
        className="c0 c1"
        fill="currentColor"
        version="1.1"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M12,20A8,8 0 0,1 4,12A8,8 0 0,1 12,4A8,8 0 0,1 20,12A8,8 0 0,1 12,20M12,2A10,10 0 0,0 2,12A10,10 0 0,0 12,22A10,10 0 0,0 22,12A10,10 0 0,0 12,2Z"
          fillRule="evenodd"
        />
      </svg>
    </div>
    <input
      checked={false}
      className="input_field accessibly_hidden"
      onChange={[MockFunction]}
      type="radio"
      value="hazelnut"
    />
    <div
      className="label_text"
    >
      Hazelnut
    </div>
  </label>
  <label
    className="form_field"
  >
    <div
      className="checkbox_icon"
    >
      <svg
        aria-hidden="true"
        className="c0 c1"
        fill="currentColor"
        version="1.1"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M12 20a8 8 0 0 1-8-8 8 8 0 0 1 8-8 8 8 0 0 1 8 8 8 8 0 0 1-8 8m0-18A10 10 0 0 0 2 12a10 10 0 0 0 10 10 10 10 0 0 0 10-10A10 10 0 0 0 12 2m0 5a5 5 0 0 0-5 5 5 5 0 0 0 5 5 5 5 0 0 0 5-5 5 5 0 0 0-5-5z"
          fillRule="evenodd"
        />
      </svg>
    </div>
    <input
      checked={true}
      className="input_field accessibly_hidden"
      onChange={[MockFunction]}
      type="radio"
      value="chocolate"
    />
    <div
      className="label_text"
    >
      Chocolate
    </div>
  </label>
  <label
    className="form_field"
  >
    <div
      className="checkbox_icon"
    >
      <svg
        aria-hidden="true"
        className="c0 c1"
        fill="currentColor"
        version="1.1"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M12,20A8,8 0 0,1 4,12A8,8 0 0,1 12,4A8,8 0 0,1 20,12A8,8 0 0,1 12,20M12,2A10,10 0 0,0 2,12A10,10 0 0,0 12,22A10,10 0 0,0 22,12A10,10 0 0,0 12,2Z"
          fillRule="evenodd"
        />
      </svg>
    </div>
    <input
      checked={false}
      className="input_field accessibly_hidden"
      onChange={[MockFunction]}
      type="radio"
      value="ginger"
    />
    <div
      className="label_text"
    >
      Ginger
    </div>
  </label>
</div>
`;

exports[`RadioGroup renders correctly with checked value 1`] = `
.c0 {
  width: 100%;
}

.c1.spin {
  -webkit-animation: GLFYz 0.8s steps(8) infinite;
  animation: GLFYz 0.8s steps(8) infinite;
  -webkit-transform-origin: center;
  -ms-transform-origin: center;
  transform-origin: center;
}

<div
  className=""
>
  <label
    className="form_field"
  >
    <div
      className="checkbox_icon"
    >
      <svg
        aria-hidden="true"
        className="c0 c1"
        fill="currentColor"
        version="1.1"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M12,20A8,8 0 0,1 4,12A8,8 0 0,1 12,4A8,8 0 0,1 20,12A8,8 0 0,1 12,20M12,2A10,10 0 0,0 2,12A10,10 0 0,0 12,22A10,10 0 0,0 22,12A10,10 0 0,0 12,2Z"
          fillRule="evenodd"
        />
      </svg>
    </div>
    <input
      checked={false}
      className="input_field accessibly_hidden"
      onChange={[MockFunction]}
      type="radio"
      value="hazelnut"
    />
    <div
      className="label_text"
    >
      Hazelnut
    </div>
  </label>
  <label
    className="form_field"
  >
    <div
      className="checkbox_icon"
    >
      <svg
        aria-hidden="true"
        className="c0 c1"
        fill="currentColor"
        version="1.1"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M12 20a8 8 0 0 1-8-8 8 8 0 0 1 8-8 8 8 0 0 1 8 8 8 8 0 0 1-8 8m0-18A10 10 0 0 0 2 12a10 10 0 0 0 10 10 10 10 0 0 0 10-10A10 10 0 0 0 12 2m0 5a5 5 0 0 0-5 5 5 5 0 0 0 5 5 5 5 0 0 0 5-5 5 5 0 0 0-5-5z"
          fillRule="evenodd"
        />
      </svg>
    </div>
    <input
      checked={true}
      className="input_field accessibly_hidden"
      onChange={[MockFunction]}
      type="radio"
      value="chocolate"
    />
    <div
      className="label_text"
    >
      Chocolate
    </div>
  </label>
  <label
    className="form_field"
  >
    <div
      className="checkbox_icon"
    >
      <svg
        aria-hidden="true"
        className="c0 c1"
        fill="currentColor"
        version="1.1"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M12,20A8,8 0 0,1 4,12A8,8 0 0,1 12,4A8,8 0 0,1 20,12A8,8 0 0,1 12,20M12,2A10,10 0 0,0 2,12A10,10 0 0,0 12,22A10,10 0 0,0 22,12A10,10 0 0,0 12,2Z"
          fillRule="evenodd"
        />
      </svg>
    </div>
    <input
      checked={false}
      className="input_field accessibly_hidden"
      onChange={[MockFunction]}
      type="radio"
      value="ginger"
    />
    <div
      className="label_text"
    >
      Ginger
    </div>
  </label>
</div>
`;

exports[`RadioGroup renders correctly with no checked value 1`] = `
.c0 {
  width: 100%;
}

.c1.spin {
  -webkit-animation: GLFYz 0.8s steps(8) infinite;
  animation: GLFYz 0.8s steps(8) infinite;
  -webkit-transform-origin: center;
  -ms-transform-origin: center;
  transform-origin: center;
}

<div
  className=""
>
  <label
    className="form_field"
  >
    <div
      className="checkbox_icon"
    >
      <svg
        aria-hidden="true"
        className="c0 c1"
        fill="currentColor"
        version="1.1"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M12,20A8,8 0 0,1 4,12A8,8 0 0,1 12,4A8,8 0 0,1 20,12A8,8 0 0,1 12,20M12,2A10,10 0 0,0 2,12A10,10 0 0,0 12,22A10,10 0 0,0 22,12A10,10 0 0,0 12,2Z"
          fillRule="evenodd"
        />
      </svg>
    </div>
    <input
      checked={false}
      className="input_field accessibly_hidden"
      onChange={[MockFunction]}
      type="radio"
      value="hazelnut"
    />
    <div
      className="label_text"
    >
      Hazelnut
    </div>
  </label>
  <label
    className="form_field"
  >
    <div
      className="checkbox_icon"
    >
      <svg
        aria-hidden="true"
        className="c0 c1"
        fill="currentColor"
        version="1.1"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M12,20A8,8 0 0,1 4,12A8,8 0 0,1 12,4A8,8 0 0,1 20,12A8,8 0 0,1 12,20M12,2A10,10 0 0,0 2,12A10,10 0 0,0 12,22A10,10 0 0,0 22,12A10,10 0 0,0 12,2Z"
          fillRule="evenodd"
        />
      </svg>
    </div>
    <input
      checked={false}
      className="input_field accessibly_hidden"
      onChange={[MockFunction]}
      type="radio"
      value="chocolate"
    />
    <div
      className="label_text"
    >
      Chocolate
    </div>
  </label>
  <label
    className="form_field"
  >
    <div
      className="checkbox_icon"
    >
      <svg
        aria-hidden="true"
        className="c0 c1"
        fill="currentColor"
        version="1.1"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M12,20A8,8 0 0,1 4,12A8,8 0 0,1 12,4A8,8 0 0,1 20,12A8,8 0 0,1 12,20M12,2A10,10 0 0,0 2,12A10,10 0 0,0 12,22A10,10 0 0,0 22,12A10,10 0 0,0 12,2Z"
          fillRule="evenodd"
        />
      </svg>
    </div>
    <input
      checked={false}
      className="input_field accessibly_hidden"
      onChange={[MockFunction]}
      type="radio"
      value="ginger"
    />
    <div
      className="label_text"
    >
      Ginger
    </div>
  </label>
</div>
`;
